firecracker/microvm.py

```diff
@@ -262,30 +267,68 @@ def enable_rootfs(self, path_on_host: str) -> str:
             rootfs_filename = Path(path_on_host).name
             jailer_path_on_host = f"/opt/{rootfs_filename}"
             os.link(path_on_host, f"{self.jailer_path}/{jailer_path_on_host}")
-            return jailer_path_on_host
+            return Path(jailer_path_on_host)
         else:
             return path_on_host
 
+    def mount_rootfs(self, path_on_host: Path) -> Path:
+        """Mount a rootfs to the VM.
+        """
+        self.mounted_fs = path_on_host
+        if not self.use_jailer:
+            return path_on_host
+
+        rootfs_filename = Path(path_on_host).name
+        jailer_path_on_host = Path(DEVICE_BASE_DIRECTORY) / rootfs_filename
+        if not (self.jailer_path / jailer_path_on_host).is_block_device():
+            device_vm_path = Path(DEVICE_BASE_DIRECTORY) / str(self.vm_id)
+            device_vm_path.mkdir(exist_ok=True, parents=True)
+            path_to_mount = device_vm_path / rootfs_filename
+            if not path_to_mount.is_symlink():
```

Comment:
We discussed using `pathlib.Path(...).resolve()` to use the target of symlinks directly and refactor this secion.